addons/purchase/models/purchase_order_line.py

```diff
@@ -558,7 +558,7 @@ def _prepare_account_move_line(self, move=False):
 
         res = {
             'display_type': self.display_type or 'product',
-            'name': self.name,
+            'name': self.name if not self.product_id.display_name or (self.product_id.display_name in self.name or self.name in self.product_id.display_name) else f"{self.product_id.display_name} {self.name}",
```

Comment:
We can not make a helper method for this in accounting on account.move.line?  (maybe there is one already)